snapcraft/internal/meta/_snap_packaging.py

```diff
@@ -260,6 +265,23 @@ def _update_yaml_with_defaults(config_data, schema):
                     )
                 )
 
+    recommended_grade = GlobalState.load(
+        filepath=global_state_file_path
+    ).get_recommended_grade()
+    if "grade" not in config_data:
+        config_data["grade"] = recommended_grade
+        logger.warning(
+            "'grade' property not specified: defaulting to {!r}.".format(
+                recommended_grade
+            )
+        )
+    elif config_data["grade"] == "stable" and recommended_grade != "stable":
+        logger.warning(
+            "The  recommended 'grade' for this snap is 'devmode' as it relies "
```

Comment:
devmode v. devel

Reply:
Perhaps simplify to `f"The  recommended 'grade' for this snap is {recommended_grade!r} as it relies on a base that is not available on the stable channel."`

Reply:
This gives no indication of why we are showing this message though (the fact that it was forced to stable in snapcraft.yaml). I will use it anyways and let you think on it :-)